core/services/pipeline/runner.go

```diff
@@ -358,3 +482,26 @@ func (r *runner) runReaper() {
 		logger.Errorw("Pipeline run reaper failed", "error", err)
 	}
 }
+
+// init task: Searches the database for runs stuck in the 'running' state while the node was previously killed.
+// We pick up those runs and resume execution.
+func (r *runner) scheduleUnfinishedRuns() {
+	// limit using a createdAt < now() @ start of run to prevent executing new jobs
+	now := time.Now()
+
+	// immediately run reaper so we don't consider runs that are too old
+	r.runReaper()
+
+	err := r.orm.GetUnfinishedRuns(now, func(run Run) error {
+		// SAFETY: ??? is this safe to do concurrently
+		go func() {
```

Comment:
Not safe IMO, needs to be added to the waitgroup on the runner and synchronised with the main loop